src/Microsoft.AspNetCore.Sockets.Http/ConnectionManager.cs

```diff
@@ -94,8 +97,19 @@ public void RemoveConnection(string id)
 
         private static string MakeNewConnectionId()
         {
-            // TODO: We need to sign and encyrpt this
-            return Guid.NewGuid().ToString();
+#if NETCOREAPP2_1
+            // 128 bit buffer / 8 bits per byte = 16 bytes
+            Span<byte> buffer = stackalloc byte[16];
+            _keyGen.GetBytes(buffer);
+            // Generate the id with RNGCrypto because we want a cryptographically random id, which GUID is not
+            return WebEncoders.Base64UrlEncode(buffer.ToArray());
```

Comment:
Convert to hex, don't base64 encode.

Reply:
Y u hate base64? It looks so much cleaner, and no ugly `==` because it's the URL encode variant.

Plus, the hex encoding code looks gross (`BitConverter.ToString(buffer.ToArray()).Replace("-", "")`)

```
Base64URL: MTIzNDU2NzgxMjM0NTY3OA
Hex:       31323334353637383132333435363738

Reply:
So much cleaner, looks more encrypty (it's "Base64 Encrypted" after all ;))

Reply:
fite me irl

Reply:
Then we need to update the Base64 encoder to support taking a span and we need to update it to use the new base64 decoder.

Reply:
https://github.com/aspnet/Home/issues/2966

Reply:
Until then, lets remove the stackalloc since that just forces us to allocate more.